Gems/WhiteBox/Code/Source/SubComponentModes/EditorWhiteBoxTransformMode.cpp

```diff
@@ -51,6 +51,17 @@ namespace WhiteBox
             buttonId);
     }
 
+    static void SetViewportUiClusterDisableButton(
+        AzToolsFramework::ViewportUi::ClusterId clusterId, AzToolsFramework::ViewportUi::ButtonId buttonId, bool isDisabled)
+    {
+        AzToolsFramework::ViewportUi::ViewportUiRequestBus::Event(
+            AzToolsFramework::ViewportUi::DefaultViewportId,
+            &AzToolsFramework::ViewportUi::ViewportUiRequestBus::Events::SetClusterDisableButton,
+            clusterId,
+            buttonId,
+            isDisabled);
+    }
+
```

Comment:
Thanks for the feedback @mbalfour-amzn, that is likely a good idea, potentially it might be clearer to actually just have two functions, `EnableClusterButton` and `DisableClusterButton` to make the callsites clearer and avoid bool params. I'll catch-up with @amzn-ahmadkrm about it today 👍

Reply:
Thanks! For something like this, I'm personally a fan of passing in the bool so that you have "EnableClusterButton(true)" / "EnableClusterButton(false)" vs different functions, but they both work well. The main reason I prefer a bool param is that typically the code that _calls_ an API like this has a boolean state variable already, and so it's a bit clunky to make the caller write:
```
if (enableState) 
{
    EnableClusterButton();
}
else 
{
    DisableClusterButton();
}
```
vs
```
EnableClusterButton(enableState);
```

It's overall just less boilerplate code to maintain. I find them both very readable though, so no objections from me if you go the more verbose route.

Reply:
I just always hate 'Enable' being in the name when it's not actually enabling anything, so maybe `SetClusterButtonState` might be useful (I'm having deja vu as I'm sure I've had this conversation about this very topic before 😅)